src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/OutputGroupProvider.cs

```diff
@@ -0,0 +1,95 @@
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Build
+{
+    [Export(typeof(IOutputGroupProvider))]
+    [AppliesTo(ProjectCapabilities.VisualStudioWellKnownOutputGroups)]
+    [Order(1000)]
+    internal class OutputGroupProvider : IOutputGroupProvider
+    {
+        /// <summary>
+        /// List of well known output groups names and their associated target and description
+        /// </summary>
+        private static readonly ImmutableHashSet<IOutputGroup> s_outputGroups = ImmutableHashSet.Create<IOutputGroup>()
+            .Add(NewGroup("PublishItems", "PublishItemsOutputGroup", Resources.OutputGroupPublishItemsDisplayName, Resources.OutputGroupPublishItemsDescription));
+
+        /// <summary>
+        /// The set of output groups that apply to this project.
+        /// </summary>
+        private readonly AsyncLazy<IImmutableSet<IOutputGroup>> _outputGroups;
+
+        private readonly IProjectAccessor _projectAccessor;
+        private readonly ConfiguredProject _configuredProject;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OutputGroupProvider"/> class.
+        /// </summary>
+        /// <param name="projectAccessor">Imported <see cref="IProjectAccessor"/>.</param>
+        /// <param name="configuredProject">Imported <see cref="ConfiguredProject"/>.</param>
+        [ImportingConstructor]
+        private OutputGroupProvider(IProjectAccessor projectAccessor, ConfiguredProject configuredProject)
+        {
+            _projectAccessor = projectAccessor;
+            _configuredProject = configuredProject;
+            _outputGroups = new AsyncLazy<IImmutableSet<IOutputGroup>>(GetOutputGroupMetadataAsync);
+        }
+
+        /// <inheritdoc/>
+        public Task<IImmutableSet<IOutputGroup>> OutputGroups
+        {
+            get { return _outputGroups.GetValueAsync(); }
+        }
+
+        /// <summary>
+        /// Gets a collection of names of targets in this project.
+        /// </summary>
+        /// <returns>Collection of the names of targets in this project.</returns>
+        private Task<ImmutableHashSet<string>> GetProjectTargetsAsync()
+        {
+            return _projectAccessor.OpenProjectForReadAsync(_configuredProject, project =>
+                project.Targets.Keys.ToImmutableHashSet(StringComparer.OrdinalIgnoreCase));
+        }
+
+        /// <summary>
+        /// Produces a set of output groups that is a subset of all well known output groups.
+        /// </summary>
+        /// <returns>Set of all known output groups.</returns>
+        private async Task<IImmutableSet<IOutputGroup>> GetOutputGroupMetadataAsync()
+        {
+            // Start with the comment set of output groups.
+            var result = s_outputGroups;
```

Comment:
Please follow var rules, there will be messages in the ErrorList telling you what to change.

Reply:
Done.  I didn't see these originally because they're messages instead of errors/warnings like the rest of the style rules, and I apparently had those turned off.